### PR TITLE
VideoPlayer: Fix reloading translation remapped stream

### LINKED ISSUE #43917: VideoPlayer: Remaps not supported
**Godot version:** v3.2.2

**OS/device including version:**
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
OpenGL ES 3.0 Renderer: GeForce RTX 2080 SUPER/PCIe/SSE2

**Issue description:**
Videos (resources) configured in remaps has no effect on VideoPlayer when the language is changed

**Steps to reproduce:**
Create VideoPlayer add Resource, create Remaps and change Languages, language change has no effect on VideoPlayer videofile.

**Minimal reproduction project:**
[VideoRemaps.zip](https://github.com/godotengine/godot/files/5607984/VideoRemaps.zip)

### PR DESCRIPTION
**Pull request by akien-mga**
- Fixes #43917.
- Reverts #43920.

MRP ported from Godot 3 (might need opening a couple of times, there's some weirdness with the font):
[VideoRemaps.zip](https://github.com/godotengine/godot/files/13328297/VideoRemaps.zip)

### Notes

- For some reason, `.ogv` extension doesn't show up in the list of resources that can be picked from translation remaps. So one needs to select "All Files" to select `.ogv` files.
- A good UX improvement would be to parse the end of the file to see if it's a locale name, and automatically pre-select the matching locale (and country if relevant), so `video_de.ogv` should automatically select "German", and `video_de_AT` should automatically select "German (Austria)".
- This changes means that if the VideoStream resource changes while it's being played, the playback will stop. It's still possible for users to handle automatically restarting the playback when that happens, with something like this (modified from MRP):

```gdscript
extends Control

var is_playing := false

func _ready() -> void:
	$Video.stream.changed.connect(_on_stream_changed, CONNECT_DEFERRED)
	_on_EN_pressed()

func _on_stream_changed() -> void:
	if is_playing:
		_on_Play_pressed()

func _on_video_finished():
	is_playing = false

func _on_EN_pressed() -> void:
	TranslationServer.set_locale("en")

func _on_DE_pressed() -> void:
	TranslationServer.set_locale("de")

func _on_RU_pressed() -> void:
	TranslationServer.set_locale("ru")

func _on_Play_pressed() -> void:
	print("pressed")
	$Video.stop()
	$Video.play()
	is_playing = true
```

But it may not be easy to seek to the previous position. To do so, we would need to only emit `changed`, but not handle in the VideoStreamPlayer, leaving it to the user to implement.

Alternatively the VideoStreamPlayer could have more signals like `stream_changed` that could include relevant information about the previous state (was playing, position, etc.). But the current player is fairly barebones and that's way outside the scope of this bugfix.

### CODE REVIEW

**Comment by KoBeWi** [comment]
> For some reason, .ogv extension doesn't show up in the list of resources that can be picked from translation remaps. So one needs to select "All Files" to select .ogv files.

I investigated and it's caused by this:
https://github.com/godotengine/godot/blob/d443f127ef1db9fb2fe623ba1c17ded5402aff10/modules/theora/video_stream_theora.cpp#L676-L678
The best part is that any resource with this method will not appear on the list. Consequently, any resource you can select does not implement `handles_type()`. I wonder what is this method for 🙃

File: scene/gui/video_stream_player.cpp [hunk 1]
```diff
@@ -237,6 +237,12 @@ bool VideoStreamPlayer::has_loop() const {
 void VideoStreamPlayer::set_stream(const Ref<VideoStream> &p_stream) {
 	stop();
 
+	// Make sure to handle stream changes seamlessly, e.g. when done via
+	// translation remapping.
```
Comment (KoBeWi) [inline scene/gui/video_stream_player.cpp:241]: Shouldn't this comment be below?

**Review by KoBeWi — APPROVED** [review]

File: scene/gui/video_stream_player.cpp (hunk 1 above)
Comment (akien-mga) [reply scene/gui/video_stream_player.cpp:241]: It was meant to cover both at once, but since they're split it's not very clear yeah.

**Comment by YuriSizov** [comment]
Cherry-picked for 4.2.1.

**Pull request merged**